Sources/NIOOpenSSL/OpenSSLHandler.swift

```diff
@@ -151,8 +151,10 @@ public class OpenSSLHandler : ChannelInboundHandler, ChannelOutboundHandler {
         case .closing:
             // We're in the process of TLS shutdown, so let's let that happen. However,
             // we want to cascade the result of the first request into this new one.
-            if let promise = promise {
-                closePromise!.futureResult.cascade(promise: promise)
+            if let promise = promise, let closePromise = self.closePromise {
```

Comment:
should we maybe refactor:

```swift
if let promise = promise {
    if let closePromise = self.closePromise {
        closePromise.futureResult.cascade(promise: promise)
    } else {
        self.closePromise = promise
    }
}
```

Reply:
I thought about it, but decided against it. In particular, I don't believe the compiler will generate better code under this refactor: the compiler should be aware that optional unwrapping does not have side effects, so it can evaluate the unwrap only once. If the compiler won't generate more efficient code, I think we should go for clarity, and avoiding the extra indentation seems clearer to me.

Reply:
That said, @weissi may disagree.

Reply:
tbh I don't think better/worse code matters here. But yes, the compiler will know that unwrapping an optional doesn't have side-effects. I'd also go for clarity here.

Reply:
I actually think the suggestion is clearer ... but I am not strong here so works for me :)

Reply:
maybe @Lukasa should toss a coin :P

Reply:
My proposal is that I find the current construction clearer, and as I spend more time in this codebase than @normanmaurer that might be more important ;)

Reply:
works for me

Reply:
@Lukasa haha... can't disagree with this ;) Like I said I am not strong here so ignore me buddy